feat: add DocumentAIProcessor as a direct resource

jingyih opened the pull request.

yuwenma commented on pkg/test/resourcefixture/testdata/basic/documentai/v1alpha1/documentaiprocessor/_http.log:
Would you mind uploading the real GCP log?

jingyih replied:
PR updated. Real GCP log is here: https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/3557/commits/e7ec0169ec0243de569100bdc282ca426afcd7b0

yuwenma reviewed (commented):
/approve

Awesome! One comment

yuwenma reviewed (commented):
/lgtm

jingyih commented:
/hold

jingyih commented:
Rebase to resolve merge conflict.

yuwenma commented:
Did you create the version "pretrained-ocr-v1.2-2022-11-10" between step 1 and 2?

yuwenma commented on apis/documentai/v1alpha1/processor_types.go:
```diff
@@ -0,0 +1,146 @@
+	// +kcc:proto:field=google.cloud.documentai.v1.Processor.display_name
+	DisplayName *string `json:"displayName,omitempty"`
+
+	// The default processor version.
+	// +kcc:proto:field=google.cloud.documentai.v1.Processor.default_processor_version
+	DefaultProcessorVersion *string `json:"defaultProcessorVersion,omitempty"`
```
Can we remove this field for now?

jingyih replied:
Done
https://github.com/GoogleCloudPlatform/k8s-config-connector/commit/eb7c57121ffe7d00ef5ae989fa11abf03926d20c

yuwenma reviewed (commented):
/approve

one suggestion, otherwise look good

jingyih commented:
Rebased to resolve merge conflict. Also added https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/3557/commits/eb7c57121ffe7d00ef5ae989fa11abf03926d20c to address comments on removing the `DefaultProcessorVersion` field.

yuwenma commented:
/lgtm

jingyih commented:
/hold cancel
